monashspa/PHS3302/d0_lifetime/d0_lifetime.py

```diff
@@ -0,0 +1,410 @@
+
+ax[0,1].hist(D0_Y,bins=100)
+ax[0,1].set_xlabel('$y_{decay}$ [mm]')
+ax[0,1].set_ylabel('Candidates')
+
+ax[0,2].hist(D0_Z,bins=100)
+ax[0,2].set_xlabel('$z_{decay}$ [mm]')
+ax[0,2].set_ylabel('Candidates')
+
+ax[1,0].hist(PV_X,bins=100)
+ax[1,0].set_xlabel('$x_{production}$ [mm]')
+ax[1,0].set_ylabel('Candidates')
+
+ax[1,1].hist(PV_Y,bins=100)
+ax[1,1].set_xlabel('$y_{production}$ [mm]')
+ax[1,1].set_ylabel('Candidates')
+
+ax[1,2].hist(PV_Z,bins=100)
+ax[1,2].set_xlabel('$z_{production}$ [mm]')
+ax[1,2].set_ylabel('Candidates')
+
+plt.tight_layout()
+plt.show()
+
+
+# ## Part 2: Creating the information we'll need
+# 
+# The data set contains the three components of momentum  of the kaon and pion, and the positions of their vertex and the proton-proton interaction vertex. From this we will need to create the information we need: the mass and decay time of the $D^0$ candidate.  
+# 
+# ### Part 2a: $D^0$ mass
+# The data contains the 3-momenta of the kaon and pion:
+# $$
+# \vec{\mathbf{p}} = (p_{x},p_{y},p_{z})
+# $$
+# We need the 4-momenta, which means adding the energy of the particles as well. At the LHCb experiment we don't measure the particle energy directly. Instead we use particle identification detectors to determine what type of particle the various the long-lived particles are (e.g. kaons or pions) and use the externally measured values of their masses and our measured momentums to compute the energy using 
+# $$
+# E^2 = |\vec{\mathbf{p}}|^2 + m^2
+# $$
+# 
+# Values of the pion and kaon masses have been measured elswhere:
+# - $m(\pi) = 139.6\,{\rm MeV}/c^{2}$
+# - $m(K) = 493.7\,{\rm MeV}/c^{2}$ 
+# 
+# **Task 1:** *[2 marks]* Using the example below, create a new variable called `K_E` and `Pi_E` in which you calculate the pion and kaon energies from the 3-momenta and their known masses.
+
+
+
+# Pi_E = 
+# K_E = 
+
+
+# Now we have the pion and kaon energies we have the complete 4-momentum for the kaon and pion
+# 
+# $$ 
+# \mathbf{p}_{\pi} = \left[E_\pi,p_{x}(\pi),p_{y}(\pi),p_{z}(\pi)\right]
+# $$
+# $$
+# \mathbf{p}_{K} = \left[E(K),p_{x}(K),p_{y}(K),p_{z}(K)\right]
+# $$
+# we can use these to calculate the mass of the $D^0$ meson candidate. 
+# 
+# 
+# The mass can be calculated from the sum of the momentum 4-vectors for the kaon and pion 
+# $$
+# \mathbf{p}_{D^{0}} = \mathbf{p}_{K} + \mathbf{p}_{\pi},
+# $$
+# 
+# In the rest frame of the $D^0$ meson we will have $\mathbf{p}_{D^0} = \left[m_{D^0},0,0,0\right]$
+# $$
+# m_{D^{0}}^2 = \mathbf{p}_{D^{0}}.\mathbf{p}_{D^{0}} = (\mathbf{p}_{K} + \mathbf{p}_{\pi})^2.
+# $$
+# **Task 2:** *[2 marks]* Calculate the $D^0$ mass using the all of the components of the kaon and pion 4-momentum. 
+
+
+
+## D0_M = 
+
+
+# A loose selection has already been applied to the sample we are analysing. This means that the signal contains both real $D^0\to K \pi$ decays as well as backgrounds from combinations of a kaon and pion that didn't originate from a $D^0$ meson. 
```

Comment:
You start by introducing the unknown word selection, and then explain it afterwards. Maybe turn around.

Reply:
Thanks, I've reworded this bit